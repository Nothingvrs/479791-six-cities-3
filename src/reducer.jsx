import offers from './mocks/offers';

export const ActionType = {
  CHANGE_CITY: `change-city`,
  GET_OFFER: `get-offer`,
  CLEAR_OFFER: `clear-offer`
};

export const getOffersByCity = (initialOffers, cityName) =>
  initialOffers.filter((offer) => offer.city === cityName);

export const getCities = (initialOffers) => {
  const cities = new Set();
  initialOffers.forEach((offer) => cities.add(offer.city));
  return [...cities];
};

export const getOfferById = (initialOffers, id) => {
  const offer = initialOffers.find((newOffer) => newOffer.id === Number(id));
  if (!offer) {
    return null;
  }
  const newOffer = Object.assign({}, offer);
  newOffer.nearOffers = offer.nearOffers.map((offerId) =>
    initialOffers.find((offerItem) => offerItem.id === offerId)
  );
  return newOffer;
};

const initialState = {
  city: getCities(offers)[0],
  offers: getOffersByCity(offers, getCities(offers)[0]),
  citiesNames: getCities(offers)
};

export const ActionCreator = {
  changeCity: (cityName) => {
    const newOffers = getOffersByCity(offers, cityName);

    return {type: ActionType.CHANGE_CITY, payload: {offers: newOffers, city: cityName}};
  },
  getCardDetails(id) {

    const newOffer = getOfferById(offers, id);

    return {type: ActionType.GET_OFFER, payload: newOffer};
  },
  clearOffer() {
    return {type: ActionType.CLEAR_OFFER, payload: ``};
  }
};

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.CHANGE_CITY:
      return Object.assign({}, state, {city: action.payload.city, offers: action.payload.offers});
    case ActionType.GET_OFFER:
      return Object.assign({}, state, {offer: action.payload});
    case ActionType.CLEAR_OFFER:
      return Object.assign({}, state, {offer: action.payload});
  }

  return state;
};
