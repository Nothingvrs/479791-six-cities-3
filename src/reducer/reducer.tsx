import {combineReducers} from "redux";
import {userReducer as user} from './user/user-reducer';
import {dataReducer as data} from "./data/data-reducer";
const rootReducer = combineReducers({
  user,
  data
});

export default rootReducer;
