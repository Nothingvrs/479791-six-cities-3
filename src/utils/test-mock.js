export const mockTestData = {
  numberOfOffers: 32,
  cards: [
    `Canal View Prinsengracht`,
    `Nice, cozy, warm big bed apartment`,
    `Wood and stone place`
  ]
};

export const mockCards = [
  {
    id: 0,
    name: `Wood and stone place`,
    type: `Private room`,
    imgs: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`],
    price: 35,
    isInBookmark: false,
    mark: 2,
    isPremium: false,
    bedroomNo: 1,
    capacity: 1,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,

    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 5,
    avgMark: 5,
    hostUser: {
      name: `Angelina`,
      img: `img/avatar-angelina.jpg`,
      status: `pro`
    },
    addressCoords: [52.3909553943508, 4.929309666406198]
  },
  {
    id: 1,
    name: `Beautiful & luxurious apartment at great location `,
    type: `Private room`,
    imgs: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`],
    price: 555,
    isInBookmark: true,
    mark: 3,
    isPremium: false,
    bedroomNo: 1,
    capacity: 1,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`
    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Angelina`,
      img: `img/avatar-angelina.jpg`,
      status: `pro`
    },
    addressCoords: [52.369553943508, 4.85309666406198]
  },
  {
    id: 2,
    name: `Canal View Prinsengracht`,
    type: `Apartment`,
    imgs: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`],
    price: 132,
    isInBookmark: false,
    mark: 4,
    isPremium: false,
    bedroomNo: 1,
    capacity: 1,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`
    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Angelina`,
      img: `img/avatar-angelina.jpg`,
      status: `pro`
    },
    addressCoords: [52.3909553943508, 4.929309666406198]
  },
  {
    id: 3,
    name: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    imgs: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`],
    price: 180,
    isInBookmark: false,
    mark: 5,
    isPremium: false,
    bedroomNo: 1,
    capacity: 1,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`
    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Angelina`,
      img: `img/avatar-angelina.jpg`,
      status: `pro`
    },
    addressCoords: [52.3809553943508, 4.939309666406198]
  }
];


export const findByTestAtr = (component, attr) => {
  return component.find(`[data-test='${attr}']`);
};


