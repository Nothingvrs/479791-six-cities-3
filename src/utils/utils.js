import PropTypes from "prop-types";

export const cardPropTypes = PropTypes.shape({
  name: PropTypes.string.isRequired,
  type: PropTypes.oneOf([`Apartment`, `Private room`]).isRequired,
  imgs: PropTypes.arrayOf(PropTypes.string).isRequired,
  price: PropTypes.number.isRequired,
  isInBookmark: PropTypes.bool.isRequired,
  mark: PropTypes.oneOf([...new Array(6)].map((_, i) => i)).isRequired,
  isPremium: PropTypes.bool.isRequired,
  bedroomNo: PropTypes.number.isRequired,
  capacity: PropTypes.number.isRequired,
  facilities: PropTypes.arrayOf(PropTypes.string).isRequired,
  descriptions: PropTypes.arrayOf(PropTypes.string).isRequired,
  reviews: PropTypes.number.isRequired,
  avgMark: PropTypes.number.isRequired,
  hostUser: PropTypes.shape({
    name: PropTypes.string.isRequired,
    img: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired
  }).isRequired}).isRequired;
