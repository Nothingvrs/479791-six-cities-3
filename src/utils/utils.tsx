export interface CommentModel {
  id: number;
  comment: string;
  mark: number;
  date: string;
  user: UserModel;
}

interface HostUserModel {
  name: string;
  img: string;
  isPro: boolean;
}

interface LocationModel {
  latitude: number;
  longitude: number;
  zoom: number;
}

export interface UserModel {
  id: number;
  email: string;
  name: string;
  img: string;
  isPro: boolean;
}

export interface CityModel {
  name?: string;
  location: LocationModel;
}

export interface CardModel {
  id: number;
  name: string;
  type: string;
  previewImg: string;
  imgs: string [];
  price: number;
  isInBookmark: boolean;
  mark: number;
  isPremium: boolean;
  bedroomNo: number;
  capacity: number;
  facilities: string [];
  descriptions: string [];
  reviews: number;
  avgMark: number;
  comments: CommentModel [];
  hostUser: HostUserModel;
  city: CityModel;
  addressCoords: [number, number];
  nearOffers?: CardModel [];
}


export const offerAdapter = (offer) => {
  return {
    id: offer.id,
    name: offer.title,
    type: offer.type.charAt(0).toUpperCase() + offer.type.slice(1),
    previewImg: offer.preview_image,
    imgs: offer.images,
    price: offer.price,
    isInBookmark: offer.is_favorite,
    mark: offer.rating,
    isPremium: offer.is_premium,
    bedroomNo: offer.bedrooms,
    capacity: offer.max_adults,
    facilities: offer.goods,
    descriptions: [offer.description],
    hostUser: {
      id: offer.host.id,
      name: offer.host.name,
      img: offer.host.avatar_url,
      isPro: offer.host.is_pro
    },
    addressCoords: [offer.location.latitude, offer.location.longitude],
    city: offer.city
  };
};

export const commentAdapter = (comment) => ({
  id: comment.id,
  comment: comment.comment,
  mark: comment.rating,
  date: comment.date,
  user: userAdapter(comment.user)
});

export const userAdapter = (user) => ({
  id: user.id,
  email: user.email,
  name: user.name,
  img: user.avatar_url,
  isPro: user.is_pro
});

export const getCities = (initialOffers) => {
  const citiesNames = new Set();
  const cities = [];
  initialOffers.forEach((offer) => {
    if (!citiesNames.has(offer.city.name)) {
      citiesNames.add(offer.city.name);
      cities.push(offer.city);
    }
  });
  return cities;
};
