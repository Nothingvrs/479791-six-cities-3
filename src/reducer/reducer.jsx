import {offerAdapter} from '../utils/utils';

export const getCities = (initialOffers) => {
  const citiesNames = new Set();
  const cities = [];
  initialOffers.forEach((offer) => {
    if (!citiesNames.has(offer.city.name)) {
      citiesNames.add(offer.city.name);
      cities.push(offer.city);
    }
  });
  return cities;
};

export const Action = {
  SET_HOVERED: `set-hovered`,
  RESET_HOVERED: `reset-hovered`,
  SET_CITY: `set-city`,
  SET_FILTER: `set-filter`,
  RESET_FILTER: `reset-filter`,
  GET_OFFERS: `get-offers`
};

export const ActionCreator = {
  setHovered(id) {
    return {type: Action.SET_HOVERED, payload: id};
  },
  resetHovered() {
    return {type: Action.RESET_HOVERED};
  },
  setFilter(filterName) {
    return {type: Action.SET_FILTER, payload: filterName};
  },
  resetFilter() {
    return {type: Action.RESET_FILTER};
  },
  setCity(cityName) {
    return {type: Action.SET_CITY, payload: cityName};
  },
  getOffersFromApi() {
    return (dispatch, state, api) => {
      api.get(`/hotels`).then((response) => {
        dispatch({
          type: Action.GET_OFFERS,
          payload: response.data.map((offer) => offerAdapter(offer))
        });
      });
    };
  }
};

const initialState = {
  filterName: `popular`,
  hoveredId: -1
};

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case Action.SET_HOVERED:
      return Object.assign({}, state, {hoveredId: action.payload});
    case Action.RESET_HOVERED:
      return Object.assign({}, state, {hoveredId: -1});
    case Action.SET_FILTER:
      return Object.assign({}, state, {filterName: action.payload});
    case Action.RESET_FILTER:
      return Object.assign({}, state, {filterName: `popular`});
    case Action.SET_CITY:
      return Object.assign({}, state, {city: action.payload});
    case Action.GET_OFFERS:
      return Object.assign({}, state, {
        offers: action.payload,
        city: getCities(action.payload)[0],
        citiesNames: getCities(action.payload),
        isLoaded: true
      });
  }
  return state;
};
