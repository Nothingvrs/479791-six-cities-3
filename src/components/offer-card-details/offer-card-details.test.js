import OfferCardDetails from './offer-card-details.jsx';
import React from 'react';
import {mockCards} from '../../utils/test-mock';
import Enzyme, {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import EnzymeReactAdapter from 'enzyme-adapter-react-16';
import {getCities, getOffersByCity} from '../../reducer';
import {Provider} from 'react-redux';
import {createStore} from 'redux';

Enzyme.configure({adapter: new EnzymeReactAdapter()});

const initialState = {
  city: getCities(mockCards)[0],
  offers: getOffersByCity(mockCards, getCities(mockCards)[0]),
  citiesNames: getCities(mockCards)
};

const reducer = (state = initialState) => {
  return state;
};
const store = createStore(reducer);

Enzyme.configure({adapter: new EnzymeReactAdapter()});

it(`OfferCardDetails successfully rendered`, () => {
  const mockHistory = {push: jest.fn};
  const mockMatch = {
    params: {
      id: 0
    }
  };
  const tree = mount(
      <Provider store={store}>
        <OfferCardDetails
          card={mockCards[0]}
          onHeaderClick={() => {}}
          history={mockHistory}
          match={mockMatch}
        />
      </Provider>
  );
  expect(toJson(tree, {mode: `deep`})).toMatchSnapshot();
});
