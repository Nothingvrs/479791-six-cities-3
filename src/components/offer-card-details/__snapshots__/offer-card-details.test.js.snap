// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferCardDetails successfully rendered 1`] = `
<Connect(OfferCardDetail)
  card={
    Object {
      "addressCoords": Array [
        52.3909553943508,
        4.85309666406198,
      ],
      "avgMark": 4.5,
      "bedroomNo": 1,
      "capacity": 1,
      "city": Object {
        "location": Object {
          "latitude": 52.37454,
          "longitude": 4.897976,
          "zoom": 13,
        },
        "name": "Amsterdam",
      },
      "comments": Array [
        Object {
          "author": "Max",
          "authorImg": "/img/avatar-max.jpg",
          "comment": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
          "date": "2020-01-26T13:51:50.417Z",
          "id": 1,
          "mark": 4,
        },
        Object {
          "author": "Max",
          "authorImg": "/img/avatar-max.jpg",
          "comment": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
          "date": "2020-01-26T13:51:50.417Z",
          "id": 2,
          "mark": 4,
        },
      ],
      "descriptions": Array [
        "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
        "An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.",
      ],
      "facilities": Array [
        "Wi-Fi",
        "Heating",
        "Kitchen",
        "Fridge",
        "Washing machine",
        "Coffee machine",
        "Dishwasher",
      ],
      "hostUser": Object {
        "img": "/img/avatar-angelina.jpg",
        "isPro": true,
        "name": "Angelina",
      },
      "id": 0,
      "imgs": Array [
        "/img/apartment-01.jpg",
        "/img/apartment-02.jpg",
        "/img/apartment-03.jpg",
        "/img/studio-01.jpg",
        "/img/room.jpg",
      ],
      "isInBookmark": true,
      "isPremium": true,
      "mark": 4,
      "name": "Beautiful & luxurious apartment at great location",
      "nearOffers": Array [
        1,
        2,
        3,
      ],
      "previewImg": "/img/apartment-01.jpg",
      "price": 120,
      "reviews": 2,
      "type": "Apartment",
    }
  }
  history={
    Object {
      "push": [Function],
    }
  }
  match={
    Object {
      "params": Object {
        "id": 0,
      },
    }
  }
  onHeaderClick={[Function]}
>
  <div
    className="page"
  >
    <header
      className="header"
    >
      <div
        className="container"
      >
        <div
          className="header__wrapper"
        >
          <div
            className="header__left"
          >
            <a
              className="header__logo-link"
              href="/"
            >
              <img
                alt="6 cities logo"
                className="header__logo"
                height="41"
                src="/img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            className="header__nav"
          >
            <ul
              className="header__nav-list"
            >
              <li
                className="header__nav-item user"
              >
                <Link
                  className="header__nav-link header__nav-link--profile"
                  to="/login"
                >
                  <LinkAnchor
                    className="header__nav-link header__nav-link--profile"
                    href="/login"
                    navigate={[Function]}
                  >
                    <a
                      className="header__nav-link header__nav-link--profile"
                      href="/login"
                      onClick={[Function]}
                    >
                      <div
                        className="header__avatar-wrapper user__avatar-wrapper"
                      />
                      <span
                        className="header__login"
                      >
                        Sign in
                      </span>
                    </a>
                  </LinkAnchor>
                </Link>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      className="page__main page__main--property"
    >
      <section
        className="property"
      >
        <div
          className="property__gallery-container container"
        >
          <div
            className="property__gallery"
          >
            <div
              className="property__image-wrapper"
              key="0"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="/img/apartment-01.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
              key="1"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="/img/apartment-02.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
              key="2"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="/img/apartment-03.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
              key="3"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="/img/studio-01.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
              key="4"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="/img/room.jpg"
              />
            </div>
          </div>
        </div>
        <div
          className="property__container container"
        >
          <div
            className="property__wrapper"
          >
            <div
              className="property__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              className="property__name-wrapper"
            >
              <h1
                className="property__name"
              >
                Beautiful & luxurious apartment at great location
              </h1>
              <button
                className="property__bookmark-button button property__bookmark-button--active"
                type="button"
              >
                <svg
                  className="property__bookmark-icon"
                  height="33"
                  width="31"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  In
                   bookmarks
                </span>
              </button>
            </div>
            <div
              className="property__rating rating"
            >
              <div
                className="property__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "90%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
              <span
                className="property__rating-value rating__value"
              >
                4.5
              </span>
            </div>
            <ul
              className="property__features"
            >
              <li
                className="property__feature property__feature--entire"
              >
                Apartment
              </li>
              <li
                className="property__feature property__feature--bedrooms"
              >
                1
                 Bedroom
              </li>
              <li
                className="property__feature property__feature--adults"
              >
                Max 
                1
                 adult
              </li>
            </ul>
            <div
              className="property__price"
            >
              <b
                className="property__price-value"
              >
                €
                120
              </b>
              <span
                className="property__price-text"
              >
                 night
              </span>
            </div>
            <div
              className="property__inside"
            >
              <h2
                className="property__inside-title"
              >
                What's inside
              </h2>
              <ul
                className="property__inside-list"
              >
                <li
                  className="property__inside-item"
                  key="0"
                >
                  Wi-Fi
                </li>
                <li
                  className="property__inside-item"
                  key="1"
                >
                  Heating
                </li>
                <li
                  className="property__inside-item"
                  key="2"
                >
                  Kitchen
                </li>
                <li
                  className="property__inside-item"
                  key="3"
                >
                  Fridge
                </li>
                <li
                  className="property__inside-item"
                  key="4"
                >
                  Washing machine
                </li>
                <li
                  className="property__inside-item"
                  key="5"
                >
                  Coffee machine
                </li>
                <li
                  className="property__inside-item"
                  key="6"
                >
                  Dishwasher
                </li>
              </ul>
            </div>
            <div
              className="property__host"
            >
              <h2
                className="property__host-title"
              >
                Meet the host
              </h2>
              <div
                className="property__host-user user"
              >
                <div
                  className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
                >
                  <img
                    alt="Host avatar"
                    className="property__avatar user__avatar"
                    height="74"
                    src="//img/avatar-angelina.jpg"
                    width="74"
                  />
                </div>
                <span
                  className="property__user-name"
                >
                  Angelina
                </span>
              </div>
              <div
                className="property__description"
              >
                <p
                  className="property__text"
                  key="0"
                >
                  A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
                </p>
                <p
                  className="property__text"
                  key="1"
                >
                  An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.
                </p>
              </div>
            </div>
          </div>
        </div>
        <section
          className="property__map map"
        />
      </section>
      <div
        className="container"
      >
        <section
          className="near-places places"
        >
          <h2
            className="near-places__title"
          >
            Other places in the neighbourhood
          </h2>
          <div
            className="near-places__list places__list"
          >
            Array [
              <article
                className="near-places__card place-card"
                data-test="test-card-hover"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="near-places__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height="200"
                      src="/img/apartment-01.jpg"
                      width="260"
                    />
                  </a>
                </div>
                <div
                  className="place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        80
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button place-card__bookmark-button--active"
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        In
                         bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "90%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      data-test="test-header-click"
                      href="#"
                      onClick={[Function]}
                    >
                      Wood and stone place
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    Private room
                  </p>
                </div>
              </article>,
              <article
                className="near-places__card place-card"
                data-test="test-card-hover"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="near-places__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height="200"
                      src="/img/apartment-01.jpg"
                      width="260"
                    />
                  </a>
                </div>
                <div
                  className="place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        132
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button false"
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To
                         bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "90%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      data-test="test-header-click"
                      href="#"
                      onClick={[Function]}
                    >
                      Canal View Prinsengracht
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    Apartment
                  </p>
                </div>
              </article>,
              <article
                className="near-places__card place-card"
                data-test="test-card-hover"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="near-places__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height="200"
                      src="/img/apartment-01.jpg"
                      width="260"
                    />
                  </a>
                </div>
                <div
                  className="place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        180
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button false"
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To
                         bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "90%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      data-test="test-header-click"
                      href="#"
                      onClick={[Function]}
                    >
                      Nice, cozy, warm big bed apartment
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    Apartment
                  </p>
                </div>
              </article>,
            ]
          </div>
        </section>
      </div>
    </main>
  </div>
</Connect(OfferCardDetail)>
`;
